shared/src/main/kotlin/watchfaces/dslclock/DSLClockComplication.kt

```diff
@@ -0,0 +1,355 @@
+                textLayoutResult = textLayoutResult,
+                topLeft = Offset(
+                    x = 60f,
+                    y = (center.y * 0.4f) - (textLayoutResult.size.height / 2f)
+                )
+            )
+        }
+    }
+}
+
+
+@Composable
+private fun Date(style: TextStyle) {
+
+    val formatter = DateTimeFormatter.ofPattern("MMM dd, yyyy")
+    val date = LocalDate.now().format(formatter)
+
+    @OptIn(InternalComposeOClockApi::class)
+    val measurer = LocalTextMeasurerWithoutCache.current
+
+    val textLayoutResult by remember {
+        derivedStateOf {
+            measurer.measure(
+                "date = $date",
+                style
+            )
+        }
+    }
+
+    OClockCanvas {
+        rotate(degrees = 0f) {
+            drawText(
+                brush = style.brush ?: SolidColor(style.color),
+                textLayoutResult = textLayoutResult,
+                topLeft = Offset(
+                    x = 120f,
+                    y = (center.y * 0.6f) - (textLayoutResult.size.height / 2f)
+                )
+            )
+        }
+    }
+
+}
+
+
+@Composable
+private fun Battery(style: TextStyle) {
+
+    val context: Context = LocalContext.current
+    var batteryLevel by remember { mutableIntStateOf(0) }
+
+    val iFilter = IntentFilter(Intent.ACTION_BATTERY_CHANGED)
+    val batteryStatus = context.registerReceiver(null, iFilter)
+
+    batteryLevel = batteryStatus?.getIntExtra(BatteryManager.EXTRA_LEVEL, -1) ?: -1
+
+    context.registerReceiver(
+        object : BroadcastReceiver() {
+            override fun onReceive(ctxt: Context, intent: Intent) {
+                try {
+                    batteryLevel = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1)
+                } catch (_: IllegalStateException) {
+                }
+            }
+        },
+        IntentFilter(Intent.ACTION_BATTERY_CHANGED))
+
+    @OptIn(InternalComposeOClockApi::class)
+    val measurer = LocalTextMeasurerWithoutCache.current
+
+    val textLayoutResult by remember {
+        derivedStateOf {
+            measurer.measure(
+                "battery = $batteryLevel / 100",
+                style
+            )
+        }
+    }
+
+    OClockCanvas {
+        rotate(degrees = 0f) {
+            drawText(
+                brush = style.brush ?: SolidColor(style.color),
+                textLayoutResult = textLayoutResult,
+                topLeft = Offset(
+                    x = 120f,
+                    y = (center.y * 0.8f) - (textLayoutResult.size.height / 2f)
+                )
+            )
+        }
+    }
+
+}
+
+
+@Composable
+private fun HeartRate(style: TextStyle) {
+
+    val context = LocalContext.current
+    val isAmbient by LocalIsAmbient.current
+    var hr by remember { mutableIntStateOf(0) }
+
+    val sensorManager = context.getSystemService(Context.SENSOR_SERVICE) as SensorManager
+    val hrCounter = sensorManager.getDefaultSensor(Sensor.TYPE_HEART_RATE)
```

Comment:
Same as above in `Battery`, but we probably want to replace that with use of Complications API or WHS PassiveClient anyway.